src/ProjectTemplates/Shared/DevelopmentCertificate.cs

```diff
@@ -1,42 +1,61 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System;
-using System.IO;
-using Microsoft.AspNetCore.Certificates.Generation;
+using System.Reflection;
+using System.Text.Json;
 
 namespace Templates.Test.Helpers;
 
-public readonly struct DevelopmentCertificate
+public readonly struct DevelopmentCertificate(string certificatePath, string certificatePassword, string certificateThumbprint)
 {
-    public DevelopmentCertificate(string certificatePath, string certificatePassword, string certificateThumbprint)
+    public readonly string CertificatePath { get; } = certificatePath;
+    public readonly string CertificatePassword { get; } = certificatePassword;
+    public readonly string CertificateThumbprint { get; } = certificateThumbprint;
+
+    public static DevelopmentCertificate Get(Assembly assembly)
     {
-        CertificatePath = certificatePath;
-        CertificatePassword = certificatePassword;
-        CertificateThumbprint = certificateThumbprint;
-    }
+        string[] locations = [
+            Path.Combine(AppContext.BaseDirectory, "aspnetcore-https.json"),
+            Path.Combine(Environment.CurrentDirectory, "aspnetcore-https.json"),
+            Path.Combine(Path.GetDirectoryName(assembly.Location)!, "aspnetcore-https.json")
```

Comment:
Can you add a comment explaining why the json file could end up in any of these three directories? It's only ever generated by PrepareForTest.targets, right?

Reply:
They are all the same 😄 

I was trying to figure out Helix as I couldn't find the cert (it wasn't being copied). I'll just use `appcontext.basedirectory`